--- originalPlay.c	2017-11-13 10:41:25.000000000 -0200
+++ play.c	2017-11-13 11:24:29.004510653 -0200
@@ -46,16 +46,47 @@
    issuing a single 'write' call for each line. Would this yield any significant
    performance improvement? */
 
+int firstTime = 1;
 void draw (char scene[NSCENES][NROWS][NCOLS], int number)
 {
-  int i, j;
-  for (i=0; i<NROWS; i++)
-    {
-      for (j=0; j<NCOLS; j++)
-	{
-	  putchar (scene[number][i][j]);
-	}
-      putchar ('\n');
-      putchar ('\r');
+  int i, j, s;
+  /* Each line will have this length */
+  int totalLength = (NROWS*(NCOLS+2) + 1);
+  /* Array of strings */
+  char strings[NSCENES+1][totalLength];
+  int position;
+  if(firstTime != 0) {
+    /* Go through all the scenes */
+    for (s = 0; s < NSCENES; s++) {
+      position = 0;
+      for (i = 0; i < NROWS; i++) {
+        for (j = 0; j < NCOLS; j++) {
+          strings[s][position++] = scene[s][i][j];
+        }
+        /* Add special characteres */
+        strings[s][position++] = '\n';
+        strings[s][position++] = '\r';
+      }
+      /*  End of string */
+      strings[s][position++] = '\0';
     }
+    /* Print the required string  */
+    printf("%s", strings[number]);
+    firstTime = 0;
+  } else {
+    /* This scene presents problems when being displayed using just printf, so for this case, i'll do the normal way */
+    if (number == 7) {
+      for (i = 0; i < NROWS; i++) {
+          for (j = 0; j < NCOLS; j++) {
+            putchar (scene[number][i][j]);
+          }
+          /* Add special characteres */
+          putchar ('\n');
+          putchar ('\r'); 
+      }
+    } else {
+      /* Print the required string  */
+      printf("%s", strings[number]);
+    }
+  } 
 }
